scripts/secure-boot/generate-images/generate-image.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+SCRIPTDIR=$(readlink -f $(dirname "$0"))
+COUNT=$1
+TESTNAME=$2
+N=$3
+
+# go into a separate directory to make cleanup easier
+TEMPDIR=$(mktemp -d)
+cd "$TEMPDIR"
+
+# make the new image
+dd if=/dev/zero of=image.img bs=1M count=$COUNT
+mkfs.fat -F 12 image.img
+
+if [ "$TESTNAME" = "INTERMEDIATE" ]; then
+
+    # generate the first key pair
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/" -keyout cert.key -out cert.crt -days 3650 -nodes -sha256
+
+    # generate the second key pair
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/" -keyout PK.key -out PK.crt -days 3650 -nodes -sha256
+
+    # make the CSR, its necessary to input something (country=PL, company=3mdeb)
+    openssl req -new -key cert.key -out cert.csr -subj "/C=PL/O=3mdeb"
+
+    # sign the CSR with the second key pair
+    # its necessary to `touch cert.ext`, otherwise it says its not there
+    touch cert.ext
+    openssl x509 -req -in cert.csr -CA PK.crt -CAkey PK.key -out cert.crt -CAcreateserial -extfile cert.ext
+    openssl x509 -in cert.crt -outform der -out cert.der
+
+elif [ "$TESTNAME" = "RSA" ]; then
+
+    openssl req -new -x509 -newkey rsa:$N -subj "/CN=PK/" -keyout cert.key -out cert.crt -days 3650 -nodes -sha256
+    openssl x509 -in cert.crt -outform der -out cert.der
+
+elif [ "$TESTNAME" = "ECDSA256" ]; then
```

Comment:
@WiktorG351 are you sure that we can't prepare one function for those cases?

Reply:
@TomaszAIR this was added in [this commit](https://github.com/Dasharo/open-source-firmware-validation/pull/192/commits/73b95c5b65c10a70a68646905f85f706c358f235)